numpy/_core/fromnumeric.py

```diff
@@ -2643,6 +2668,35 @@ def all(a, axis=None, out=None, keepdims=np._NoValue, *, where=np._NoValue):
                                   keepdims=keepdims, where=where)
 
 
+def _cumulative_sum_dispatcher(x, /, *, axis=None, dtype=None, out=None,
+                               include_initial=None):
+    return (x, out)
+
+
+@array_function_dispatch(_cumulative_sum_dispatcher)
+def cumulative_sum(x, /, *, axis=None, dtype=None, out=None,
+                   include_initial=False):
+    """
+    Array API compatible cumulative_sum
```

Comment:
Since this is presumably going to be exposed in the main namespace, this needs a proper docstring, including the differences with regular `cumsum`.

Reply:
Done!